### ISSUE TITLE
Fix sizing of `QGridWidget`

### ISSUE THREAD

**Issue by ellisonbg** [quantopian/qgrid#44]
Right now the `QGridWidget` sizes itself in wonky ways. It actually extends beyond that edge of the widget area. The main thing is to setup the sizing such that:
- It works inside other widgets that use flexbox for layout.
- It works at the top level and stays within the widget area.

ping @aggFTW

**Comment by ssanderson** [comment]
ping @TimShawver @blink1073.  Do you guys have a sense of what would need to change to make our sizing behavior more well-behaved?

**Comment by ellisonbg** [comment]
I have some ideas. In general, the widget-area div has the flex box
information. From a width perspective, the QGridWidget can set its flexbox
attributes to just grow to fit its parent rather than sizing itself in px.
Although you might want to size the height in px. We are making changes to
ipywidget that will improve the access to the flexbox stuff on the python
side. I might be able to take a crack at this.

On Mon, Nov 2, 2015 at 7:52 PM, Scott Sanderson <notifications@github.com>
wrote:

> ping @TimShawver <https://github.com/TimShawver> @blink1073
> <https://github.com/blink1073>. Do you guys have a sense of what would
> need to change to make our sizing behavior more well-behaved?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/quantopian/qgrid/issues/44#issuecomment-153207157>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by aggFTW** [comment]
@ellisonbg, what code were you using when you ran into this issue?

I'm currently doing this. Notice that I skipped using the QGrid widget and I'm just using display() to render the pandas df:
```
def on_render_table(*args):
        with output:
            output.clear_output()

            data = viz.data
            if isinstance(data, pd.Series):
                df = pd.DataFrame(data)
            else:
                df = data
            # display(get_qgrid_widget(df))

            display(data)
```

Both `display(get_gqrid_widget(df)` and `display(data)` exceed the width of the widget -- the qgrid widget actually displays in the bottom while the pandas df exceeds the width.

The only difference I see between the two calls is that output.clear_ouput() has no effect for the qgrid_widget; it just appends the new one below.

How should we be rendering them inside output to avoid such problems?

**Comment by aggFTW** [comment]
For the `display(data)` one, I see this:

<img width="1255" alt="screen shot 2015-11-02 at 5 12 00 pm" src="https://cloud.githubusercontent.com/assets/654204/10898998/1d4e748c-8185-11e5-9896-bd29592a1cd1.png">

**Comment by aggFTW** [comment]
After clicking two times on Table (which activates the `on_render_table` I showed) the `display(get_qgrid_widget(df)` shows this (notice the duplicated widget):

<img width="794" alt="screen shot 2015-11-02 at 5 12 58 pm" src="https://cloud.githubusercontent.com/assets/654204/10899003/330e3bea-8185-11e5-9113-f66b33857f0b.png">

**Comment by ellisonbg** [comment]
For the overall widget you will want to next everything in a heirarchy of
HBox ad VBox widgets and they toggle the visible attribute to hide and show
parts of it based on the button clicks. By doing that you can control the
visibility at a very fine grained level. You can't get this with display
and clear_output.

But that side of things is separate for the sizing of the QGridWidget.

On Mon, Nov 2, 2015 at 8:15 PM, Alejandro Guerrero Gonzalez <
notifications@github.com> wrote:

> After clicking two times on Table (which activates the on_render_table I
> showed) the display(get_qgrid_widget(df) shows this (notice the
> duplicated widget):
>
> [image: screen shot 2015-11-02 at 5 12 58 pm]
> <https://cloud.githubusercontent.com/assets/654204/10899003/330e3bea-8185-11e5-9113-f66b33857f0b.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/quantopian/qgrid/issues/44#issuecomment-153209981>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by blink1073** [comment]
I'm afraid I'm of little use here.  What I did was a straight up hack, and I've dealt very little with CSS.

**Comment by TimShawver** [comment]
I think this might be fixed in the new qgrid 1.0 beta, but it's hard to tell because this is so old that I imagine a lot has changed in the notebook since it was filed (and a lot has changed in qgrid too).  I'm going to close it since I'm not really sure how to test it...please open a new issue if someone finds a repro case in which qgrid 1.0  is not sizing itself correctly.
